### DISCUSSION TITLE
Pytorch deep learning video time stamp 7hr 10 min

### DISCUSSION THREAD

**Discussion by c2vishal** [mrdbourke/pytorch-deep-learning#1345] [General]
while plotting the list(epoch_count,loss_values,test_loss) in graph  daniel was getting error to convert tensor values in numpy arr
when i used inference_mode() for my plot to graph and i dont need to do the conversion is this the right approach?
code:
 #plot the loss curves
with torch.inference_mode():
  plt.plot(epoch_count,loss_values,label="Train loss")
  plt.plot(epoch_count,test_loss_values,label="Test loss")
  plt.title("Training and test loss curves")
  plt.ylabel("Loss")
  plt.xlabel("Epochs")

**Comment by Avalianfeng** [comment]
When tensor computation is not performed using a GPU, the generated `loss_values` and `test_loss_values` are directly processed on the CPU and can be used directly by matplotlib.

Therefore, there's no need to use `np.array(torch.tensor(loss_values).numpy())`.

You can directly pass in:
`loss_value = loss_function(y_pred, y_train)`
`loss_values.append(loss.item())`

However, you must use `item()` for processing.

When your tensor contains only one element (e.g., a scalar `loss`), calling `.item()` will do two things:

Remove gradients and device: Regardless of whether the tensor is on a GPU or CPU, and regardless of whether it has gradients.

Convert the type: Directly convert this standard PyTorch scalar into a pure Python number (usually float or int).

Because Python's native float lists can be perfectly recognized and plotted by Matplotlib, using `.item()` is extremely efficient.
